site/public/js/user-profile.js

```diff
@@ -121,6 +128,50 @@ function updateUserProfilePhoto () {
   return false;
 }
 
+function updateUserSecondaryEmail () {
+    const second_email = $('#user-secondary-email-change');
+    const second_email_notify = $('#user-secondary-email-notify-change');
+    if (second_email.data('current-second-email') === second_email.val() && second_email_notify.get(0).checked === (second_email_notify.data('current-second-email-notify') === 1)) {
+        displayErrorMessage('No changes detected to secondary email');
+    }
+    else {
+        let data = new FormData();
+        data.append('csrf_token', csrfToken);
+        data.append('secondary_email', second_email.val());
+        data.append('secondary_email_notify', second_email_notify.get(0).checked);
+        let url = buildUrl(['user_profile', 'change_secondary_email']);
+        $.ajax({
+            url,
+            type: "POST",
+            data,
+            processData: false,
+            contentType: false,
+            success: function(res) {
+                const response = JSON.parse(res);
+                if (response.status === "success") {
+                    const { data } = response;
+                    displaySuccessMessage(data.message);
+                    $('#secondary-email-row .value').text(data.secondary_email);
+                    $('#secondary-email-notify-row .value').text(data.secondary_email_notify);
+                    second_email.data('current-second-email', data.secondary_email);
+                    second_email_notify.data('current-second-email-notify', data.secondary_email_notify === "True" ? 1 : 0);
+                }
+                else if (response.status === "error") {
+                    displayErrorMessage("Some went wrong while updating secondary email address!");
+                    second_email.val(second_email.data('current-second-email'));
+                    second_email_notify.prop('checked', second_email_notify.data('current-second-email-notify'));
```

Comment:
Shouldn't reset the fields on error as that makes it harder for user to correct a typo or whatever they had wrong on submit.

Reply:
I was debating this myself. The reason I went for this was so it doesn't look like that is still saved in there. Maybe I should make a message appear saying UNSAVED changes without changing the actual data back. Thoughts?

Reply:
[`setCustomValidity`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLObjectElement/setCustomValidity) may be a good option to show an error message tooltip on the input?

Reply:
I changed the input box to type email so the browser will not let you submit the form until that matches a valid email. I removed these lines as well since that really shouldn't be necessary.